### PR TITLE
Add Test Info panel

### PR DESCRIPTION
**Pull request by jaril**
This exposes the metadata for tests within a spec and its error (if applicable). [PR for reference](https://github.com/replayio/replay-cli/pull/90).

For Playwright ([example](http://localhost:8080/recording/should-close-modal-on-escape--f3dc7a93-06af-4dd7-b143-a0922f53c0d7)), this shows a spec file with one test.
<details>
<summary>Screenshot</summary>
image
</details>

For Cypress ([example](http://localhost:8080/recording/cypresse2eadding-spects--536ee87b-9db7-49c2-b65e-20e0d0d1a5d0?point=10384593718841091478773302472737184&time=4131.571612903226&hasFrames=false)), this shows a spec file with multiple tests.<details>
<summary>Screenshot</summary>
image
</details>

### CODE REVIEW

File: src/ui/components/SidePanel.tsx [hunk 1]
```diff
@@ -34,12 +35,20 @@ export default function SidePanel() {
     },
   ];
 
-  if (cypressResults && cypressResults.length > 0) {
+  console.log({ recording });
```
Comment (ryanjduffy) [inline src/ui/components/SidePanel.tsx:38]: 🪵

File: src/ui/types/index.ts [hunk 2]
```diff
@@ -158,11 +158,27 @@ export type TestMetadata = {
...
   file?: string;
 };
 
+export type TestItem = {
+  id?: string;
+  title: string;
+  path?: string[];
+  relativePath?: string;
+  result: TestResult;
+  error: TestItemError;
```
Comment (ryanjduffy) [inline src/ui/types/index.ts:173]: This should be optional

File: src/devtools/client/debugger/src/components/TestInfo/TestInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    <div className="flex flex-col gap-1">
+      <button className="flex flex-row items-center gap-1 group" onClick={onClick}>
+        <Status result={test.result} />
+        <div className="overflow-hidden whitespace-pre overflow-ellipsis group-hover:underline">
+          {test.title}
+        </div>
+      </button>
+      {test.error ? (
+        <div className="font-mono ml-7">
+          <div className="overflow-hidden overflow-ellipsis">{test.error.message}</div>
```
Comment (ryanjduffy) [inline src/devtools/client/debugger/src/components/TestInfo/TestInfo.tsx:61]: I think you want "truncate" for the className. If we do, we should make it toggleable so the user can see the full error message.

File: src/devtools/client/debugger/src/components/TestInfo/TestInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  const cypressResults = useFetchCypressSpec();
+
+  return (
+    <div className="flex flex-col px-4 py-2 space-y-1">
+      <div className="flex justify-between">
+        <div className="flex items-center gap-1">
+          {result ? <Status result={result} /> : null}
+          <div className="font-bold">{spec}</div>
+        </div>
+        {recordingDuration ? <div>{getFormattedTime(recordingDuration)}</div> : null}
```
Comment (ryanjduffy) [inline src/devtools/client/debugger/src/components/TestInfo/TestInfo.tsx:30]: This text is not inline with the test title. Adding `items-center` to L25 will do the trick

image

**Review by ryanjduffy — CHANGES_REQUESTED** [review]
Looks good! The thrown error is the only blocking fix to me.

**Comment by jasonLaster** [comment]
Preview branch examples
* Playwright [example](https://devtools-git-1024-test-info-recordreplay.vercel.app/recording/should-close-modal-on-escape--f3dc7a93-06af-4dd7-b143-a0922f53c0d7?point=8762000948935449963767763061178368&time=6386.953177257525&hasFrames=false)

**Review by ryanjduffy — APPROVED** [review]

**Pull request merged**